bindings/pyroot_experimental/PyROOT/python/ROOT/_application.py

```diff
@@ -78,3 +78,13 @@ def init_graphics(self):
             self._inputhook_config()
 
         self._set_display_hook()
+
+    def add_infrastructure():
+        """Add infrastructure if imported by Jupyter Kernel"""
+
+        is_ipython = hasattr(__builtins__, '__IPYTHON__') or 'IPython' in sys.modules
```

Comment:
I would not put this code here, even though I understand your rational was that the application handles interactive usage.

The application is created lazily, when the first real lookup happens, not when `import ROOT` is done. And JupyROOT needs to be imported when `import ROOT` is done. So I would rather not put the import of JupyROOT in the application.

You can just put this piece of code in `ROOT/__init__.py`, right before the creation of the facade. Also, what does it mean "infrastructure" in this case? I would just place a comment above the code that says "# Configuration for usage from Jupyter notebooks"